src/nbl/video/ILogicalDevice.cpp

```diff
@@ -7,10 +7,36 @@
 using namespace nbl;
 using namespace nbl::video;
 
+static void debloatShaders(const asset::ISPIRVDebloater& debloater, std::span<const asset::IPipelineBase::SShaderSpecInfo> shaderSpecs, core::smart_refctd_ptr<asset::IShader>* outShaders, asset::IPipelineBase::SShaderSpecInfo* outShaderSpecInfos)
+{
+    for (const auto& shaderSpec: shaderSpecs)
+    {
+        const auto entryPoint = asset::ISPIRVDebloater::EntryPoint{
+            .name = shaderSpec.entryPoint,
+            .shaderStage = shaderSpec.stage,
+        };
+        if (shaderSpec.shader)
+        {
+          const auto* shader = shaderSpec.shader;
+          *outShaders = core::make_smart_refctd_ptr<asset::IShader>(debloater.debloat(shader->getCode(), std::span(&entryPoint, 1)), shader->getContentType(), std::string(shader->getFilepathHint()));
+        } else
+        {
+          *outShaders = nullptr;
+        }
+        auto debloatedShaderSpec = shaderSpec;
+        debloatedShaderSpec.shader = outShaders->get();
+        *outShaderSpecInfos = debloatedShaderSpec;
+
+        outShaders++;
+        outShaderSpecInfos++;
```

Comment:
this would have crashed your creation code, you're taking a pointer to shader spec infos by reference from the creation params and scrolling by one